checker/jdk/index/src/java/lang/String.java

```diff
@@ -1732,7 +1732,7 @@ private int lastIndexOfSupplementary(int ch, int fromIndex) {
      * @return  the index of the first occurrence of the specified substring,
      *          or {@code -1} if there is no such occurrence.
      */
-    public @GTENegativeOne @LTEqLengthOf("this") int indexOf(String str) {
+    public @GTENegativeOne @LTEqLengthOf("this") @SubstringIndexFor(value="this",offset="#1.length()-1") int indexOf(String str) {
```

Comment:
Why are the `@GTENegativeOne` and `@LTEqLengthOf` annotations needed?  Doesn't the `@SubstringIndexFor` annotation imply them, so that they can be omitted from the JDK source code for brevity and clarity?
(Same comment for other library annotations.)

Reply:
Currently it doesn't. Implementing that for `@GTENegativeOne` should be easy.